File: android/app/src/main/java/com/alandaitch/anden/data/model/Arrival.kt
package com.alandaitch.anden.data.model

import com.alandaitch.anden.util.GeoPoint
import java.time.Instant

// Modelo de dominio de un arribo para la UI. No es el JSON crudo.
data class Arrival(
    val id: String,
    val serviceId: String?,
    val lineId: Int,
    val line: TrainLine,
    val ramalName: String,
    val destinationName: String,
    val originName: String,
    val trackName: String?,
    val scheduled: Instant?,
    val estimated: Instant?,
    val secondsUntil: Int,
    val delay: DelayStatus,
    val trainLocation: GeoPoint?,
    val equipmentName: String?,
    val isElectric: Boolean,
    val isCancelled: Boolean,
    val direction: Int,
    val stateName: String?,
    val route: List<RouteStop>
)

File: Shared/Models/Arrival.swift
import Foundation
import CoreLocation

// Modelo de dominio de un arribo para la UI. No es el JSON crudo.
struct Arrival: Identifiable {
    let id: String
    let serviceId: String?
    let lineId: Int
    let line: TrainLine
    let ramalName: String
    let destinationName: String
    let originName: String
    let trackName: String?
    let scheduled: Date?
    let estimated: Date?
    let secondsUntil: Int
    let delay: DelayStatus
    let trainLocation: CLLocationCoordinate2D?
    let equipmentName: String?
    let isElectric: Bool
    let isCancelled: Bool
    let direction: Int
    let stateName: String?
    let route: [RouteStop]
}

// ServiceAlert vive en su propio archivo; ver ServiceAlert.swift.
